### PR TITLE
Update entrypoint.sh

### PR DESCRIPTION
**Pull request by luizvaz**
Allow setting custom values to the `smb.conf` 

```
server min protocol = ${CUSTOM_SMB_PROTO}NT1|SMB2
ntlm auth = ${CUSTOM_SMB_AUTH}
```

Where :
`CUSTOM_SMB_PROTO` = `NT1|SMB2|SMB3|...`  [(Value List)](https://www.samba.org/samba/docs/current/man-html/smb.conf.5.html#CLIENTMAXPROTOCOL)
`CUSTOM_SMB_AUTH` = `no|yes`

This solves issues for some clients trying to connect using an old protocol, and receiving messages like:
- reply_negprot: No protocol supported !
- ntlm_password_check: NTLMv1 passwords NOT PERMITTED

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -5,7 +5,9 @@ LOG_LEVEL="${LOG_LEVEL:-info}"
 SET_PERMISSIONS="${SET_PERMISSIONS:-false}"
 SHARE_NAME="${SHARE_NAME:-TimeMachine}"
 CUSTOM_AFP_CONF="${CUSTOM_AFP_CONF:-false}"
+CUSTOM_SMB_AUTH="${CUSTOM_SMB_AUTH:-no}"
```
Comment (mbentley) [inline entrypoint.sh:8]: Based on the current config from my time machine container, it looks like this should default to `ntlmv2-only`:

```
# testparm -v | grep "ntlm auth"
Load smb config files from /etc/samba/smb.conf
Loaded services file OK.
Weak crypto is allowed

Server role: ROLE_STANDALONE

Press enter to see a dump of your service definitions

        ntlm auth = ntlmv2-only
```

**Review by mbentley — CHANGES_REQUESTED** [review]
Thanks!  Could you also add docs to the https://github.com/mbentley/docker-timemachine/blob/master/README.md#optional-variables-for-smb section to describe the feature?

File: entrypoint.sh (hunk 1 above)
Comment (luizvaz) [reply entrypoint.sh:8]: `yes` and `no` are aliases, accordingly to Samba dos:
https://www.samba.org/samba/docs/current/man-html/smb.conf.5.html#idm7319

**Comment by luizvaz** [comment]
> Thanks! Could you also add docs to the https://github.com/mbentley/docker-timemachine/blob/master/README.md#optional-variables-for-smb section to describe the feature?

Done!

File: entrypoint.sh (hunk 1 above)
Comment (mbentley) [reply entrypoint.sh:8]: Ah, gotcha.  Thanks!  I should have RTFM.

**Review by mbentley — APPROVED** [review]
LGTM!

**Comment by luizvaz** [comment]
Tested on an image pulled and working fine.
Thanks!

**Pull request merged**